kuka_rsi_hw_interface/src/robot_control_node.cpp

```diff
@@ -22,66 +22,129 @@
 #include "kuka_rsi_hw_interface/robot_control_node.hpp"
 #include "kroshu_ros2_core/ROS2BaseLCNode.hpp"
 
-
 namespace kuka_rsi_hw_interface
 {
-
 RobotControlNode::RobotControlNode(
   const std::string & node_name,
   const rclcpp::NodeOptions & options)
 : kroshu_ros2_core::ROS2BaseLCNode(node_name, options)
-{}
+{
+  auto qos = rclcpp::QoS(rclcpp::KeepLast(1));
+  auto callback =
+    [this](sensor_msgs::msg::JointState::SharedPtr msg)
+    {this->commandReceivedCallback(msg);};
+
+  joint_command_subscription_ = this->create_subscription<sensor_msgs::msg::JointState>(
+    "rsi_joint_command", qos, callback);
 
-rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn RobotControlNode::
-on_configure(
-  const rclcpp_lifecycle::State &)
+  joint_state_publisher_ =
+    this->create_publisher<sensor_msgs::msg::JointState>("rsi_joint_state", 1);
+  joint_command_msg_ = std::make_shared<sensor_msgs::msg::JointState>();
+}
+
+CallbackReturn RobotControlNode::on_configure(const rclcpp_lifecycle::State &)
 {
-  kuka_rsi_hw_interface_ = std::make_unique<KukaHardwareInterface>(this->shared_from_this());
-  kuka_rsi_hw_interface_->configure();
+  registerParameter<std::string>(
+    "rsi_ip_address_", "127.0.0.1",
+    kroshu_ros2_core::ParameterSetAccessRights{true, true, false, false},
+    [this](const std::string & rsi_ip_address)
+    {return this->onRSIIPAddressChange(rsi_ip_address);});
+
+  registerParameter<int>(
+    "rsi_port_", 59152,
+    kroshu_ros2_core::ParameterSetAccessRights{true, true, false, false},
+    [this](int rsi_port)
+    {return this->onRSIPortAddressChange(rsi_port);});
+
+  registerParameter<int>(
+    "n_dof_", 6,
+    kroshu_ros2_core::ParameterSetAccessRights{true, true, false, false},
+    [this](int n_dof)
+    {return this->onNDOFChange(n_dof);});
+
+  kuka_rsi_hw_interface_ = std::make_unique<KukaHardwareInterface>(
+    rsi_ip_address_, rsi_port_,
+    n_dof_);
   return SUCCESS;
 }
-
-rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn RobotControlNode::
-on_activate(
-  const rclcpp_lifecycle::State &)
+CallbackReturn RobotControlNode::on_activate(const rclcpp_lifecycle::State &)
 {
-  kuka_rsi_hw_interface_->start();
+  kuka_rsi_hw_interface_->start(
```

Comment:
I would remove the joint_command argument and set it in the node after the start function